test/JDBC/expected/table_variable_xact_errors_isolation_snapshot.out

```diff
@@ -170,3 +170,171 @@ date#!#date
 DROP TYPE empDates
 GO
 
+-------------------------------------------------------------------------------
+-- BABEL-4225: Error inside function should not cause crash
+-------------------------------------------------------------------------------
+CREATE SCHEMA [Control]
+GO
+
+CREATE FUNCTION [Control].[csf_script_delete_row] (
+   @SetIDLocal  INT,
+   @InternalID  VARCHAR(64)
+) RETURNS VARCHAR(MAX)
+AS
+BEGIN
+   DECLARE @Template          VARCHAR(MAX);
+   DECLARE @Results           VARCHAR(MAX);
+   DECLARE @DeletePredicates  VARCHAR(MAX);
+   DECLARE @Ordinal           INT;   -- NOTE: In this circumstance, each column is either an update or a predicate but not both.
+   DECLARE @Columns
+   TABLE   ( Ordinal          INT IDENTITY(1, 1) NOT NULL,
+             ColumnPredicate  VARCHAR(MAX)           NULL,
+             PRIMARY KEY ( Ordinal )
+           );
+   SET @Template = 'DELETE [SchemaName].[TableName] WHERE DeleteConditionPredicates;';
+   INSERT INTO @Columns (ColumnPredicate )
+   SELECT '[' + LRV.ColumnName + '] ' + ISNULL(('= ' + NULLIF(LRV.Expression, 'NULL')), 'IS NULL')
+   FROM   [Control].[DataRow] AS RLR
+          INNER JOIN [Control].[DataRowValue] AS LRV
+          ON (RLR.SetID      = LRV.SetID         AND
+              RLR.InternalID = LRV.InternalIDRow    )
+          INNER JOIN [Control].[cvw_local_column_base] AS STC
+          ON (RLR.SchemaName = STC.SchemaName AND
+              RLR.TableName  = STC.TableName  AND
+              LRV.ColumnName = STC.ColumnName    )
+   WHERE  RLR.SetID      =  @SetIDLocal
+   AND    RLR.InternalID =  @InternalID
+   AND    LRV.MatchType  != 'N'
+   ORDER BY STC.ColumnID ASC;END; -- [Control].[csf_script_delete_row]
+go
+
+SELECT Control.csf_script_delete_row('1', ' ')
+GO
+~~START~~
+varchar
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: relation "master_control.datarow" does not exist)~~
+
+
+DROP FUNCTION [Control].[csf_script_delete_row]
+GO
+
+DROP SCHEMA [Control]
+GO
+
+
+-------------------------------------------------------------------------------
+-- BABEL-4226: Error while table and index are open.
+-------------------------------------------------------------------------------
+CREATE FUNCTION dbo.foo() RETURNS @tab TABLE (a INT PRIMARY KEY, b VARCHAR(MAX) NOT NULL) AS
+BEGIN
+        INSERT @tab(a, b) VALUES (1, NULL)
+        RETURN
+END
+GO
+
+SELECT * FROM dbo.foo()
+GO
+~~START~~
+int#!#varchar
+~~ERROR (Code: 515)~~
+
+~~ERROR (Message: null value in column "b" of relation "@tab_1" violates not-null constraint)~~
+
+
+DROP FUNCTION dbo.foo()
+GO
+
+CREATE FUNCTION dbo.foo() RETURNS @tab TABLE (a INT PRIMARY KEY, b VARCHAR(MAX) NOT NULL) AS
```

Comment:
how is this different from above error case?

Reply:
Thanks. I will fix. The one before should not have primary key